tests/test_all.py

```diff
@@ -429,6 +433,61 @@ def test_no_compat_packages(container):
         ).is_installed
 
 
+@pytest.mark.parametrize(
+    "container",
+    ALL_CONTAINERS,
+    indirect=True,
+)
+def test_bci_eula_is_correctly_available(container):
+    """Ensure that the BCI eula exists if and only if its not a LTSS container"""
+
+    bci_license = "/usr/share/licenses/product/BCI/license.txt"
+    if OS_VERSION in ALLOWED_BCI_REPO_OS_VERSIONS and OS_VERSION in (
+        *RELEASED_SLE_VERSIONS,
+        "15.6",
+    ):
+        assert container.connection.file(bci_license).exists
+        assert (
+            "SUSE Linux Enterprise Base Container Image License"
+            in container.connection.check_output(f"head -n 1 {bci_license}")
+        )
+        return
+
+    # LTSS containers should not have the BCI license, however we currently
+    # are running tests for the bci-* base os containers which are not LTSS
+    # and still contain a BCI license. As they are out of maintenance, we
+    # need to ignore them
+    if OS_VERSION in RELEASED_SLE_VERSIONS and container in (
+        BASE_CONTAINER,
+        MINIMAL_CONTAINER,
+        MICRO_CONTAINER,
+    ):
```

Comment:
This check does not do, what you thing it should be doing. `container` is an instance of `ContainerData` and not a `DerivedContainer` or a `Container`. What you can try is this:
```suggestion
    if OS_VERSION in RELEASED_SLE_VERSIONS and container.container in (
        BASE_CONTAINER.values[0],
        MINIMAL_CONTAINER.values[0],
        MICRO_CONTAINER.values[0],
    ):
```

Reply:
yeah, I realize it's not working, nor am I sure that filtering for specific containers is the right approach. I'll test your suggestion.